src/components/Input/Input.tsx

```diff
@@ -56,9 +79,21 @@ class Input extends UIComponent<any, any> {
 
   inputRef: any
 
-  computeTabIndex = props => {
-    if (!_.isNil(props.tabIndex)) return props.tabIndex
-    if (props.onClick) return 0
+  constructor(props, context) {
+    super(props, context)
+
+    this.state = {
+      value: props.value || '',
+    }
+  }
+
+  handleChange = e => {
+    const value = _.get(e, 'target.value')
+    const { clearable } = this.props
+
+    _.invoke(this.props, 'onChange', e, { ...this.props, value })
+
+    this.setState({ value })
```

Comment:
If the user passes a `value` prop, their value will no longer be respected.  Clear should only work if the user does not have a `value` prop set.

You can paste this in an example to try it:

```jsx
import React from 'react'
import { Input } from '@stardust-ui/react'

const InputExampleIconClearableShorthand = () => (
  <Input
    clearable
    icon="search"
    placeholder="Search..."
    value='keep me'
  />
)

export default InputExampleIconClearableShorthand
```

The value of props should always be reflected in the DOM.  This is idiomatic React.  Just as when using an `input` and passing a `value` prop, the input does not update when typing in it unless you capture that event and update the prop also.  User props always win.

The AutoControlledComponent will handle this logic for you if you want to use that for now.  Otherwise, you will need to implement it yourself until we get the state manager for inputs.  I would not advise doing that as there are many edge cases (see AutoControlledComponent).